### PR TITLE
Add option for GPU and NNAPI delegates

### PR DESCRIPTION
**Pull request by thaink**
Multiple delegates can be selected now.
Models are cached in the app's local directory so NNAPI can access to them.
Screenshot_20200218-171140

### CODE REVIEW

File: cpp/backends/tflite.cc [hunk 1]
```diff
@@ -87,6 +77,43 @@ TfliteBackend::TfliteBackend(const std::string& model_file_path,
...
+    options.execution_preference =
+        tflite::StatefulNnApiDelegate::Options::kSustainedSpeed;
+    std::string accelerator_name = absl::StrContains(delegate, "-")
+                                       ? delegate.substr(delegate.find('-'))
+                                       : std::string();
+    if (!accelerator_name.empty()) {
+      options.accelerator_name = accelerator_name.c_str();
+    }
+    delegate_ptr = tflite::evaluation::CreateNNAPIDelegate(options);
+  } else if (absl::EqualsIgnoreCase(delegate, "hexagon")) {
```
Comment (terryheo) [inline cpp/backends/tflite.cc]: did you fixed the selinux error?

File: cpp/backends/tflite.cc [hunk 2]
```diff
@@ -78,32 +78,31 @@ TfliteBackend::TfliteBackend(const std::string& model_file_path,
...
+  if (absl::StartsWithIgnoreCase(delegate, "GPU")) {
     TfLiteGpuDelegateOptionsV2 gpu_opts = TfLiteGpuDelegateOptionsV2Default();
     gpu_opts.inference_preference =
         TFLITE_GPU_INFERENCE_PREFERENCE_SUSTAINED_SPEED;
-    gpu_opts.inference_priority1 = TFLITE_GPU_INFERENCE_PRIORITY_MIN_LATENCY;
-    gpu_opts.inference_priority2 =
-        TFLITE_GPU_INFERENCE_PRIORITY_MIN_MEMORY_USAGE;
-    gpu_opts.inference_priority3 = TFLITE_GPU_INFERENCE_PRIORITY_MAX_PRECISION;
-
+    if (absl::EqualsIgnoreCase(delegate, "GPU (F16)")) {
```
Comment (terryheo) [inline cpp/backends/tflite.cc]: Is this needed?

File: cpp/backends/tflite.cc [hunk 3]
```diff
@@ -78,32 +78,31 @@ TfliteBackend::TfliteBackend(const std::string& model_file_path,
...
+      gpu_opts.inference_priority1 = TFLITE_GPU_INFERENCE_PRIORITY_MIN_LATENCY;
+    } else {
+      gpu_opts.inference_priority1 =
+          TFLITE_GPU_INFERENCE_PRIORITY_MAX_PRECISION;
+    }
     delegate_ptr = tflite::evaluation::CreateGPUDelegate(&gpu_opts);
   } else if (absl::StartsWithIgnoreCase(delegate, "nnapi")) {
```
Comment (terryheo) [inline cpp/backends/tflite.cc]: Can't we use absl::StartsWith() instead?

File: java/org/mlperf/inference/res/values/array.xml [hunk 4]
```diff
@@ -3,7 +3,7 @@
   <string-array name="delegates" description="Delegate strings [CHAR_LIMIT=50]">
     <item>@string/delegate_none</item>
     <item>@string/delegate_gpu</item>
-    <item>@string/delegate_nnapi</item>
+    <item>@string/delegate_gpu_f16</item>
```
Comment (terryheo) [inline java/org/mlperf/inference/res/values/array.xml]: ditto. Isn't this require a different model?

File: cpp/proto/mlperf_task.proto [hunk 5]
```diff
@@ -59,12 +59,12 @@ message DatasetConfig {
   required DatasetType type = 2;
   // Path to the dataset.
   required string path = 3;
-  // Path to the ground truth file of the dataset.
-  // If not available, the file will be downloaded from groundtruth_src.
-  required string groundtruth_path = 4;
-  // Path to download the ground truth file if groundtruth_path not available.
-  // Could be: http, https, or @assets/file (for files embedded in assets).
-  optional string groundtruth_src = 5;
+  // Path to download the ground truth file.
```
Comment (terryheo) [inline cpp/proto/mlperf_task.proto]: Shouldn't this be a separate PR?

File: cpp/backends/tflite.cc (hunk 1 above)
Comment (thaink) [reply cpp/backends/tflite.cc]: I did. By moving the models to the app's internal directory. You can see it in my last commit.

File: cpp/backends/tflite.cc (hunk 2 above)
Comment (thaink) [reply cpp/backends/tflite.cc]: This mode is not default. So I think it is good to mention it.

File: cpp/backends/tflite.cc (hunk 3 above)
Comment (thaink) [reply cpp/backends/tflite.cc]: Users might frequently input both nnapi and NNAPI so checking both cases is required.

File: java/org/mlperf/inference/res/values/array.xml (hunk 4 above)
Comment (thaink) [reply java/org/mlperf/inference/res/values/array.xml]: No, It run both mode run on F32 models. in this mode, the accuracy is reduced to F16 so the model will run twice faster.

File: cpp/backends/tflite.cc (hunk 1 above)
Comment (thaink) [reply cpp/backends/tflite.cc]: More note: I removed hexagon, will add it when it is needed.

File: cpp/proto/mlperf_task.proto (hunk 5 above)
Comment (thaink) [reply cpp/proto/mlperf_task.proto]: It is a dependency for moving the model from /sdcard/ to the app's local directory. So I prefer to keep it here.

File: java/org/mlperf/inference/res/values/array.xml (hunk 4 above)
Comment (terryheo) [reply java/org/mlperf/inference/res/values/array.xml]: I see. I was confused with https://www.tensorflow.org/lite/performance/post_training_float16_quant
